packages/form-builder/src/settings/styles/index.tsx

```diff
@@ -0,0 +1,96 @@
+import {useState} from "react";
+import AceEditor from "react-ace";
+import debounce from 'lodash.debounce';
+import {__} from "@wordpress/i18n";
+import { fullscreen } from '@wordpress/icons';
+import {Button, Modal, PanelBody, PanelRow} from "@wordpress/components";
+
+import "ace-builds/src-noconflict/mode-css";
+import 'ace-builds/src-noconflict/snippets/css';
+import "ace-builds/src-noconflict/theme-textmate";
+import {setFormSettings, useFormState, useFormStateDispatch} from "@givewp/form-builder/stores/form-state";
+
+const CustomStyleSettings = () => {
+
+    const [ isOpen, setOpen ] = useState( false );
+    const openModal = () => setOpen( true );
+    const closeModal = () => setOpen( false );
+
+    return (
+        <PanelBody title={__('Custom Styles', 'givewp')} initialOpen={false}>
+            <PanelRow>
+
+                <div style={{
+                    width: '100%',
+                    display: 'flex',
+                    flexDirection: 'column',
+                    gap: '10px',
+                }}>
+                    <Button icon={fullscreen} onClick={ openModal }>
+                        { __( 'Edit in Modal', 'givewp' ) }
+                    </Button>
+
+                    <div style={{
+                        margin: '.5rem -16px -16px -16px', // Offset the panel padding in order to fill the inspector width.
+                    }}>
+                        <CodeControl />
+                    </div>
+
+                </div>
+
+                { !! isOpen && (
+                    <Modal overlayClassName="components-modal__screen-overlay--givewp-custom-css" title={ __( 'Custom Styles', 'givewp' ) } onRequestClose={ closeModal } style={{
+                        height: '100%',
+                        maxHeight: '100%', // Override the max height of the modal component.
+                        width: '500px',
+                        position: 'absolute',
+                        right: '0',
```

Comment:
I noticed we have a lot of inline styles going on in here, where in other places we're using SCSS.  Was this just out of convenience or necessary to override styles?

Reply:
These styles are overriding a specific instance of the modal component, so I co-located the styles with the component for context.